src/modules/schedule/controller.ts

```diff
@@ -0,0 +1,115 @@
+import { formatSchedule } from "./formatter";
+import { Schedule } from "../../generated-types/graphql";
+import { ScheduleModel } from "./model";
+import { ObjectID } from "bson";
+import { isBoolean } from "lodash";
+import { minimumViableSchedule, partialSchedule } from "./partial-schedules";
+
+
+// get the schedules for a user
+export async function getSchedulesByUser(userID: string): Promise<Schedule[]> {
+  const userSchedules = await ScheduleModel.find({created_by: userID})
+  if (userSchedules.length == 0) {
+    throw new Error("No schedules found for this user")
+  }
+  return userSchedules.map(formatSchedule)
+}
+
+// get the schedule for a user and a specific term
+export async function getScheduleByID(id: string): Promise<Schedule> {
+  const scheduleFromID = await ScheduleModel.findById({_id: id})
+  if (!scheduleFromID) {
+    throw new Error("No schedules found with this ID")
+  }
+  return formatSchedule(scheduleFromID as any)
+}
+
+// delete a schedule specified by ObjectID
+export async function removeSchedule(scheduleID: string): Promise<string> {
+  const deletedSchedule = await ScheduleModel.findByIdAndDelete(scheduleID)
+  if (!deletedSchedule) {
+    throw new Error("Schedule deletion failed")
+  }
+  return scheduleID
+}
+
+
+// create a new schedule
+export async function createSchedule(created_by: string, term: string, schedule_name: string | undefined | null, is_public: boolean, class_IDs: string[] | undefined | null, primary_section_IDs: string[], secondary_section_IDs: string[]): Promise<Schedule> {
+  
+  const newID = new ObjectID()
```

Comment:
Instead of having to do `new ObjectID()` for every new document, should let mongoose handle it by not specifying `_id` in `model.ts`. For types, extend the model type with Document.